tck-test/TestInterpreter.cpp

```diff
@@ -68,16 +67,21 @@ bool TestInterpreter::run(folly::EventBase* evb) {
 
 void TestInterpreter::handleSubscribe(const SubscribeCommand& command) {
   interactionIdToType_[command.id()] = command.type();
+  CHECK(testSubscribers_.find(command.id()) == testSubscribers_.end());
   if (command.isRequestResponseType()) {
-    auto testSubscriber = createTestSubscriber(command.id());
-    reactiveSocket_->requestResponse(
-        Payload(command.payloadData(), command.payloadMetadata()),
-        std::move(testSubscriber));
+    auto testSubscriber = make_ref<SingleSubscriber>();
+    testSubscribers_[command.id()] = testSubscriber;
+    requester_
+        ->requestResponse(
+            Payload(command.payloadData(), command.payloadMetadata()))
+        ->subscribe(std::move(testSubscriber));
```

Comment:
You have put the testSubscriber into testSubscribers_ container then you have moved testSubscriber. I wonder if it effects the object that is in the container?